### DISCUSSION TITLE
Review of documentation translation

### DISCUSSION THREAD

**Discussion by tobolar** [modelica/ModelicaStandardLibrary#4279] [Ideas]
Introducing localization for MSL (see https://github.com/modelica/ModelicaStandardLibrary/pull/4267)  will IMO require reviews of concerning PRs different to what is established now. Groups of maintainers/reviewers shall be established for every added language who look over translation modifications. So similar to library officers taking care of particular package.

And, of course, we need a new Label for it. ;-)

**Comment by HansOlsson** [comment]
We also need a lot more volunteers for it, and I believe we need a clear strategic decision regarding this before starting.

- Having so much work on translation will:
  - On one hand time spent on that will not be spent on improving the code, and we already lack reviewers etc
  - On the other hand it may attract new users and new volunteers
- Who translates?
  - I have seen volunteer translations for some programs that simply weren't good enough, and for someone like me a translation to Swedish will need to be really good to add value. Note that this depends on the language - a translation to Japanese or Mandarin might be useful even if a Swedish translation of similar quality wouldn't be useful. I agree that we need a different procedure for this, if we chose this.
  - I don't know what professional translation would cost - or how good it would be; I assume there are official price-lists.
  - With all the talk about AI it may be that a machine translation would be good enough, and a lot cheaper

**Comment by tobolar** [comment]
You are right. For Swedish people, who are generally excellent at English, there will be hardly benefit. So maybe we shall first try a language where 
1. using English is typically not that common,
2. there is a significant number of (potential) users,
3. we can establish a group of maintainers (at least at the beginning).

Maybe a tool like [weblate](https://weblate.org) may be additionally helpful here - to translate and watch the translation status as well. But I have no experience and there are surely far more possibilities

> With all the talk about AI it may be that a machine translation would be good enough, and a lot cheaper

Even then it would be good if someone checks the initial translation.

**Comment by MartinOtter** [comment]
The only practical way is fully automatic translation. The by far best translator is deepl (https://www.deepl.com),
and this is better as most manual translations - a manual check is not needed.
It has also an API where you can sent from a program a request for translation.
Its possible to provide a glossar, to translate specific terms.

The only drawback is, that the number of languages is currently 32 (asian languages are also included): 
![grafik](https://github.com/modelica/ModelicaStandardLibrary/assets/7248038/c559b63a-0b1f-4375-ae9b-60b1b39357b2)
